src/main/java/roboy/dialog/states/ordinaryStates/GamingTwentyQuestionsState.java

```diff
@@ -0,0 +1,263 @@
+package roboy.dialog.states.ordinaryStates;
+
+import com.markozajc.akiwrapper.Akiwrapper;
+import com.markozajc.akiwrapper.AkiwrapperBuilder;
+import com.markozajc.akiwrapper.core.entities.Question;
+import com.markozajc.akiwrapper.Akiwrapper.Answer;
+import com.markozajc.akiwrapper.core.entities.Guess;
+import roboy.dialog.states.definitions.State;
+import roboy.dialog.states.definitions.StateParameters;
+import roboy.linguistics.Linguistics;
+import roboy.linguistics.sentenceanalysis.IntentAnalyzer;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.ros.RosMainNode;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GamingTwentyQuestionsState extends State {
+
+
+	private static final double PROBABILITY_THRESHOLD = 0.85;
+	private final static String TRANSITION_GAME_ENDED = "gameEnded";
+
+	private Akiwrapper aw = new AkiwrapperBuilder().setFilterProfanity(true).build();
+	private Question nextQuestion = null;
+	private Guess currentGuess = null;
+
+	private List<Long> declined = new ArrayList<>();
+	private boolean userReady = false;
+	private boolean guessesAvailable = false;
+	private boolean gameFinished = false;
+	private int numberGuesses = 0;
+
+	private boolean filterApplied = false;
+	private String winner = "roboy";
+
+	public GamingTwentyQuestionsState(String stateIdentifier, StateParameters params) {
+		super(stateIdentifier, params);
+	}
+
+	@Override
+	public Output act() {
+
+		if(!userReady){
+			return Output.say("Now think of a character and tell me when you're ready to start the game.");
+
+		} else if(guessesAvailable) {
+
+			try {
+				return doAGuess();
+			}
+			catch (Exception e){
+				return Output.say("I have a problem finding a guess...");
+			}
+		}
+		else {
+
+			return askNextQuestion();
+		}
+
+	}
+
+	@Override
+	public Output react(Interpretation input) {
+
+
+		String intent = getIntent (input);
+
+		if(!isUserReady(intent)){
+
+			return Output.say("Nice, then let's start");
+
+		} else if(guessesAvailable){
+
+			return processUserGuessAnswer(intent);
+
+		} else {
+
+			try {
+
+				return saveUsersEstimate(intent);
+			}
+			catch (Exception e){
+				return Output.say("I have a problem processing your input...");
+			}
+		}
+	}
+
+	@Override
+	public State getNextState() {
+
+		if(gameFinished){
+			applyFilter(winner);
+			resetGame();
+			return getTransition(TRANSITION_GAME_ENDED);
+
+		} else {
+			return this;
+		}
+	}
+
+	private String getIntent(Interpretation input) {
+
+
+		List<String	> tokens = input.getTokens();
+
+
+
+
+
+		String intent = null;
+
+		for(String token : tokens) {
```

Comment:
should be a part of inference engine, not parsed here. 

there's also a switch operator more suitable for this case https://docs.oracle.com/javase/tutorial/java/nutsandbolts/switch.html